soroban-env-host/src/host/data_helper.rs

```diff
@@ -98,11 +98,11 @@ impl Host {
         Ok(buf)
     }
 
-    //TODO: metering!
+    // notes on metering: covers the byte cloning (write_xdr) for AlphaNum12 (worst case)
     pub fn id_preimage_from_asset(&self, asset: Asset) -> Result<Vec<u8>, HostError> {
         let pre_image = HashIdPreimage::ContractIdFromAsset(asset);
         let mut buf = Vec::new();
-        //self.charge_budget(CostType::BytesClone, 64)?; // key + salt
+        self.charge_budget(CostType::BytesClone, 44)?; // CreditAlphanum12: 12 bytes asset code + 32 bytes accountId
```

Comment:
Should we be using `size_of` to get the size of values? Ref: https://doc.rust-lang.org/std/mem/fn.size_of.html